AdminServer/appscale/admin/utils.py

```diff
@@ -497,3 +503,119 @@ def queues_from_dict(payload):
     queues[name] = queue
 
   return {'queue': queues}
+
+
+def retry_coroutine(func, async=True, backoff_base=2, backoff_multiplier=0.2,
+                    backoff_threshold=300, max_retries=None,
+                    retry_on_exception=None, refresh_args_kwargs_func=None):
+  """ Wraps func with retry mechanism which runs up to max_retries attempts
+  with exponential backoff (sleep = backoff_multiplier * backoff_base**X).
+  
+  Args:
+    func: function or coroutine to wrap.
+    async: a boolean indicating if result of function should be yielded.
+    backoff_base: a number to use in backoff calculation.
+    backoff_multiplier: a number indicating initial backoff.
+    backoff_threshold: a number indicating maximum backoff.
+    max_retries: an integer indicating maximum number of retries.
+    retry_on_exception: a function receiving one argument: exception object and
+      returning True if retry makes sense for such exception.
+    refresh_args_kwargs_func: a function without arguments which should return
+      tuple of two elements (arguments list, keyword arguments dict).
+  Returns:
+    A tornado coroutine wrapping function with retry mechanism.
+  """
+
+  @gen.coroutine
+  def persistent_execute(*args, **kwargs):
+    retries = 0
+    backoff = backoff_multiplier * (backoff_base ** 1)
```

Comment:
What's the purpose of `** 1` here?

Reply:
I believe it makes more clear that backoff_base is base for exponent.